buildkite/bazel_bench.py

```diff
@@ -0,0 +1,157 @@
+
+  Return:
+    A list of string (commit hashes).
+  """
+  day_plus_one = day + datetime.timedelta(days=1)
+  args = [
+      "git",
+      "log",
+      "--pretty=format:'%H'",
+      "--after='%s'" % day.strftime("%Y-%m-%d 00:00"),
+      "--until='%s'" % day_plus_one.strftime("%Y-%m-%d 00:00"),
+      "--reverse"
+  ]
+  command = subprocess.Popen(args, shell=True, stdoud=subprocess.PIPE)
+  return [line.strip() for line in command.stdout]
+
+
+def get_platforms(project_name):
+    """Get the platforms on which this project is run on BazelCI.
+
+    Args:
+      project_name: a string: the name of the project. e.g. "Bazel".
+
+    Returns:
+      A list of string: the platforms for this project.
+    """
+    http_config = bazelci.DOWNSTREAM_PROJECTS[project_name]["http_config"]
+    configs = bazelci.fetch_configs(http_config, None)
+    tasks = configs["tasks"]
+    return list(map(lambda k: bazelci.get_platform_for_task(k, tasks), tasks))
+
+
+def bazel_bench_for_commits(bazel_commits, platform, project):
+  """Perform bazel-bench for the platform-project combination.
+  Uploads results to BigQuery.
+
+  Args:
+    bazel_commits: a list of strings: bazel commits to be benchmarked.
+    platform: a string: the platform to benchmark on.
+    project: an object: contains the information of the project to be
+      tested on.
+
+  Return:
+    A string: the command to be executed by buildkite-agent to run bazel-bench.
+  """
+  # Get Bazel commits during the day
+  bazel_commits = get_bazel_commits(datetime.date.today())
+
+  # Download the binaries already built.
+  # Bazel-bench won"t try to build these binaries again, since they exist.
+  for bazel_commit in bazel_commits:
+    bazelci.download_bazel_binary_at_commit(
+        BAZEL_BINARY_BASE_PATH + bazel_commit,
+        platform,
+        commit
+    )
+  project_mirror_path = bazelci.get_mirror_path(
+      project["git_repository"], platform)
+  bazel_mirror_path = bazelci.get_mirror_path(BAZEL_REPOSITORY, platform)
+
+  args = [
+      "bazel",
+      "run",
+      "benchmark",
+      "--",
+      "--bazel_commits=%s" % ",".join(commits),
+      "--bazel_source=%s" % bazel_mirror_path,
+      "--project_source=%s" % project_mirror_path,
+      "--collect_memory",
+      "--runs=%s" % RUNS,
+      "--data_directory=%s" % DATA_DIRECTORY,
+      "--upload_data_to=%s" % BIGQUERY_TABLE,
+      "--",
+      project["bazel_command"]
+  ]
+
+  return " ".join(args)
+
+
+def main(argv):
+  if len(argv) > 1:
+    raise app.UsageError("Too many command-line arguments.")
+
+  bazel_bench_commands = []
+  for project in PROJECTS:
+    for platform in get_platforms(project["name"]):
+      # bazel-bench doesn't support Windows for now.
+      if platform == "windows":
+        continue
+      bazel_bench_commands.append(
+          bazel_bench_for_commits(bazel_commits, platform, project))
+
+  # Print the commands
+  print(yaml.dump({"steps": bazel_bench_commands}))
```

Comment:
you need to use bazelci.create_step() to get something that can actually run (including agent information etc)

Reply:
Done.